src/bun.js/test/expect.zig

```diff
@@ -4114,20 +4114,27 @@ pub const Expect = struct {
         if (not) pass = !pass;
         if (pass) return .undefined;
 
-        // handle failure
-        var formatter = JSC.ConsoleObject.Formatter{ .globalThis = globalObject, .quote_strings = true };
-        const received_fmt = nthCallValue.toFmt(globalObject, &formatter);
+        const received = JSValue.createEmptyArray(globalObject, nthCallValue.getLength(globalObject));
```

Comment:
JSC has a way to call `JSC::constructArray()` from a `JSC::ArgList`. `JSC::ArgList` can be constructed from a `JSC::CallFrame`. 

I also wonder if we could add a slice() binding since that's what we want here (not literally calling Array.prototype.slice but an internal function which performs a similar operation)

Maybe we should add a binding for to `JSC.CallFrame` (zig) that calls `JSC::constructArray` instead of doing this?

Reply:
The status quo is very broken, can we follow up on this?